### PR TITLE
[vSphere] Add id field to all the metricsets

### PR DESCRIPTION
**Pull request by niraj-elastic**
## Description
Added `id` field to all the metricsets.

## Checklist
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues
- Closes https://github.com/elastic/obs-integration-team/issues/73

### CODE REVIEW

File: metricbeat/module/vsphere/datastorecluster/_meta/fields.yml [hunk 1]
```diff
@@ -4,6 +4,10 @@
   description: >
     Datastore Cluster
   fields:
+    - name: id
+      type: keyword
+      description: >
+        The Datastore Cluster ID.
```
Comment (harnish-elastic) [inline metricbeat/module/vsphere/datastorecluster/_meta/fields.yml:10]: Minor nit to keep consistency

```suggestion
        Datastore Cluster ID.
```

File: metricbeat/module/vsphere/resourcepool/_meta/fields.yml [hunk 2]
```diff
@@ -26,6 +26,10 @@
               type: long
               description: >
                 The static CPU resource entitlement for a virtual machine.
+    - name: id
+      type: keyword
+      description: >
+        Resource pool ID.
```
Comment (harnish-elastic) [inline metricbeat/module/vsphere/resourcepool/_meta/fields.yml:32]:
```suggestion
        Resource Pool ID.
```

**Review by harnish-elastic — APPROVED** [review]
Minor nits. Otherwise LGTM!

**Review by lucian-ioan — APPROVED** [review]
LGTM

**Review by kush-elastic — APPROVED** [review]
LGTM

File: metricbeat/module/vsphere/host/_meta/fields.yml [hunk 3]
```diff
@@ -52,6 +52,10 @@
       description: >
         Sum of disk read and write rates each second in bytes.
       format: bytes
+    - name: id
+      type: keyword
+      description: >
+        Unique Host ID.
```
Comment (ishleenk17) [inline metricbeat/module/vsphere/host/_meta/fields.yml:58]:
```suggestion
        Unique host ID.
```

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Pull request merged**